core/api.go

```diff
@@ -316,6 +317,22 @@ func blockchainIDHandler(handler http.Handler, blockchainID string) http.Handler
 	})
 }
 
+// loggingHandler pulls out request data and adds it to the request's
+// logging context.
+func loggingHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		ctx = log.AddPrefixkv(ctx, "path", req.URL.Path)
+		if userAgent := req.UserAgent(); userAgent != "" {
+			ctx = log.AddPrefixkv(ctx, "useragent", userAgent)
```

Comment:
Instead of repeating this info on every line, do you think we could actually print a log message here and leave the log context alone? That would make subsequent log lines less verbose and maybe easier to read?